src/workers/notifications/utils.ts

```diff
@@ -64,12 +87,21 @@ export const getOptInSubscribedMembers = async ({
       status: NotificationPreferenceStatus.Subscribed,
     });
 
-  return memberQuery
+  const members = await memberQuery
     .andWhere(`EXISTS(${subscribedquery.getQuery()}) IS TRUE`)
     .getRawMany<SourceMember>();
+
+  const allowedUserIds = await filterUsersWithGlobalPreferences(
+    con,
+    members.map((m) => m.userId),
+    type,
+    'inApp',
+  );
```

Comment:
Here we introduce another query on user table, I think it should be possible to fetch notificationFlags from sourceMember by joining with user table there.